data/unaligned_labeled_mask_online_dataset.py

```diff
@@ -0,0 +1,129 @@
+import os.path
+from data.base_dataset import BaseDataset, get_transform, get_transform_seg
+from data.image_folder import make_dataset, make_labeled_mask_dataset, make_dataset_path
+from data.online_creation import crop_image, sanitize_paths
+from PIL import Image
+import random
+import numpy as np
+import torchvision.transforms as transforms
+import torch
+import torchvision.transforms.functional as F
+
+class UnalignedLabeledMaskOnlineDataset(BaseDataset):
+    """
+    This dataset class can load unaligned/unpaired datasets with mask labels.
+
+    It requires two directories to host training images from domain A '/path/to/data/trainA'
+    and from domain B '/path/to/data/trainB' respectively.
+    
+    Domain A must have labels, at the moment there are two subdirections 'images' and 'labels'.
+    
+    You can train the model with the dataset flag '--dataroot /path/to/data'.
+    Similarly, you need to prepare two directories:
+    '/path/to/data/testA' and '/path/to/data/testB' during test time.
+    """
+
+    def __init__(self, opt):
+        """Initialize this dataset class.
+
+        Parameters:
+            opt (Option class) -- stores all the experiment flags; needs to be a subclass of BaseOptions
+        """
+        BaseDataset.__init__(self, opt)
+
+        btoA = self.opt.direction == 'BtoA'
+        self.input_nc = self.opt.output_nc if btoA else self.opt.input_nc       # get the number of channels of input image
+        output_nc = self.opt.input_nc if btoA else self.opt.output_nc      # get the number of channels of output image
+
+        
+        self.dir_A = os.path.join(opt.dataroot, opt.phase + 'A')  # create a path '/path/to/data/trainA'
+        self.dir_B = os.path.join(opt.dataroot, opt.phase + 'B')  # create a path '/path/to/data/trainB'
+        if os.path.exists(self.dir_A):
+            self.A_img_paths, self.A_label_paths = make_labeled_mask_dataset(self.dir_A,'/paths.txt', opt.max_dataset_size)   # load images from '/path/to/data/trainA/paths.txt' as well as labels
```

Comment:
we're not supposed to have the labels, right ? Also the image paths are already available from the list of bbox, at least this is my understanding